src/foam/lib/parse/ReaderPStream.java

```diff
@@ -0,0 +1,86 @@
+package foam.lib.parse;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
+import java.nio.CharBuffer;
+
+public class ReaderPStream implements PStream {
+
+	protected Reference<StringReader> str;
+	protected int pos;
+	protected ReaderPStream tail_ = null;
+
+	public ReaderPStream() {
+		this(new Reference<StringReader>());
```

Comment:
The Reference was used so that you could replace the underlying String for a whole chain of StringPStreams to reuse them by just swapping out the reference, but that doesn't make sens for a Reader, so we should just get rid of the extra Reference indirection.